core/src/main/java/org/apache/calcite/rel/type/RelDataTypeFactoryImpl.java

```diff
@@ -270,12 +278,6 @@ private RelDataType copyRecordType(
       final RelRecordType type,
       final boolean ignoreNullable,
       final boolean nullable) {
-    // REVIEW: angel 18-Aug-2005 dtbug336
-    // Shouldn't null refer to the nullability of the record type
-    // not the individual field types?
-    // For flattening and outer joins, it is desirable to change
```

Comment:
The code part below can still changes the nullability of individual fields so I would suggest to keep lines 276, 277.

Reply:
Agree, I will put back lines 276, 277